libr/util/vector.c

```diff
@@ -142,7 +142,7 @@ R_API void r_vector_remove_at(RVector *vec, size_t index, void *into) {
 }
 
 R_API void *r_vector_insert(RVector *vec, size_t index, void *x) {
-	r_return_val_if_fail (vec, NULL);
+	r_return_val_if_fail (vec && index <= vec->len, NULL);
```

Comment:
Shouldnt be <?

Reply:
No, one after it is allowed in which case it will just be the same as append. The growing will always make space for at least one additional element.